### PR TITLE
Bump roslyn to latest 15.8 beta package

### PR DESCRIPTION
**Pull request by Therzok**
cc @KirillOsenkov

### CODE REVIEW

**Comment by Therzok** [comment]
seems like SRM also needs a bump

**Comment by Therzok** [comment]
@KirillOsenkov not sure why binary compat fails

**Comment by Therzok** [comment]
Also, looks like coding conventions broke on unix?

**Comment by Therzok** [comment]
So, it seems to be just unit test failures, since it tries to probe the local path of `/a.cs` from unit tests.

**Comment by Therzok** [comment]
looks green, taking a build to test it out

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/RoslynSearchCategory.cs [hunk 1]
```diff
@@ -142,25 +140,23 @@ public override Task GetResults (ISearchResultCallback searchResultCallback, Sea
...
+					var kinds = GetTagKinds (searchPattern.Tag);
 					// Maybe use language services instead of AbstractNavigateToSearchService
 					var aggregatedResults = await Task.WhenAll (TypeSystemService.AllWorkspaces
 										.Select (ws => ws.CurrentSolution)
 										.SelectMany (sol => sol.Projects)
 										.Select (async proj => {
 											using (proj.Solution.Services.CacheService?.EnableCaching (proj.Id)) {
-												var searchService = proj.LanguageServices.GetService<INavigateToSearchService> ();
+												var searchService = proj.LanguageServices.GetService<INavigateToSearchService_RemoveInterfaceAboveAndRenameThisAfterInternalsVisibleToUsersUpdate> ();
 												if (searchService == null)
```
Comment (mrward) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.MainToolbar/RoslynSearchCategory.cs]: Nice interface name `INavigateToSearchService_RemoveInterfaceAboveAndRenameThisAfterInternalsVisibleToUsersUpdate` :)

File: main/src/core/MonoDevelop.TextEditor.Tests/MonoDevelop.TextEditor.Tests.csproj [hunk 2]
```diff
@@ -1,8 +1,5 @@
 <Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="..\..\..\MonoDevelop.props" />
-  <PropertyGroup>
-    <ReferencesVSEditorCopyToOutput>true</ReferencesVSEditorCopyToOutput>
-  </PropertyGroup>
   <Import Project="$(ReferencesVSEditor)" />
```
Comment (mrward) [inline main/src/core/MonoDevelop.TextEditor.Tests/MonoDevelop.TextEditor.Tests.csproj]: Is that correct to remove?

Comment (Therzok) [reply main/src/core/MonoDevelop.TextEditor.Tests/MonoDevelop.TextEditor.Tests.csproj]: Yeah, unit test assemblies should _not_ local copy anything.

The idea is that the `TestRunner` will load [all addin assemblies](http://source.monodevelop.com/#TestRunner/Runner.cs,47). So if the addins define the right imports and dependencies, unit tests should work with just local copied GuiUnit.

Comment (Therzok) [reply main/src/core/MonoDevelop.TextEditor.Tests/MonoDevelop.TextEditor.Tests.csproj]: I had a lot of these changes in https://github.com/mono/monodevelop/pull/4528, but I had to drop the PR due to it being massive.

Maybe we can do it chunk by chunk.

**Review by mrward — APPROVED** [review]

**Comment by Therzok** [comment]
Logged non-blocking EditorConfig lib bug here: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/657362

**Comment by Therzok** [comment]
Testing check list:
* [x] Tried Main.sln and new C# projects
* [ ] Tried asp.net workloads
* [ ] Tried unity workloads
* [ ] Tried assembly browser
* [ ] Tried code issues/actions

**Comment by Therzok** [comment]
Can't find any obvious errors when doing a quick pass. Merging.

**Pull request merged**